src/Chapter2.hs

```diff
@@ -842,7 +871,14 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate _ [] = []
+rotate 0 xs = xs
+rotate a xs
+  | a < 0 = []
+  | otherwise =
+    take l (drop a (cycle xs))
```

Comment:
Nice catch! You can actually even extend it further!  It does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Using `mod` you can see a way to optimize this :slightly_smiling_face: 

(Hint: you maybe will need to join these two cases )